samza-core/src/main/java/org/apache/samza/container/RunLoop.java

```diff
@@ -276,6 +276,9 @@ private void runTasks(IncomingMessageEnvelope envelope) {
           consumerMultiplexer.tryUpdate(envelope.getSystemStreamPartition(elasticityFactor));
           log.trace("updating the system consumers for ssp keyBucket {} not processed by this runloop",
               envelope.getSystemStreamPartition(elasticityFactor));
+          // since this envelope is not processed by the container, need to decrement the # envelopes metric
+          // # envelopes metric was incremented when the envelope was returned by the SystemConsumers
+          containerMetrics.envelopes().dec();
```

Comment:
Can we add a metric here called envelopesSkipped that we increment in this case?

Reply:
added